ass_man/views.py

```diff
@@ -61,6 +64,111 @@ def get_permissions(self):
     # permission_classes = [AllowAny]
     queryset = Instance.objects.all()
 
+    def create(self, request, *args, **kwargs):
+        model_id_str = re.search(r"models/(\d+)", request.data['model'])
+        model_id = int(model_id_str.groups()[0])
+        model = Model.objects.get(pk=model_id)
+        owner_id_str = re.search(r"users/(\d+)", request.data['owner'])
+        owner_id = int(owner_id_str.groups()[0])
+        owner = User.objects.get(pk=owner_id)
+        rack_uri = request.data['rack']
+        rack_id_str = re.search(r"racks/(\d+)", rack_uri)
+        rack_id = int(rack_id_str.groups()[0])
+        rack = Rack.objects.get(pk=rack_id)
+        instance = Instance(model=model, hostname=request.data['hostname'], rack=rack, rack_u=request.data['rack_u'], owner=owner, comment=request.data['comment'])
+        instance.save()
+
+        rack_u_str = request.data['rack_u']
+        rack_u = int(rack_u_str)
+        for i in range(model.height):
+            if rack_u == 1:
```

Comment:
@rishi-tripathy I couldn't figure out how to do this any cleaner...lmk if you have any thoughts